internal/proxy/add.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# add.sh adds new dependencies to the Go Cloud module proxy. This script
+# should be run from the root of the repository.
+
+set -o pipefail
+
+function log() {
+	echo "add.sh: $@" 1>&2
+}
+
+function die() {
+	log "$@"
+	exit 1
+}
+
+if ! [[ -d .git ]]; then die 'this script must be run from the root of the repository'; fi
+
+# This approach was suggested by:
+# https://github.com/go-modules-by-example/index/tree/master/012_modvendor
+
+log "creating temp dir where we'll create a module download cache"
+tgp="$(mktemp -d)"
+if [[ $? != 0 ]]; then die 'failed'; fi
+
+temp=$(mktemp)
+if [[ $? != 0 ]]; then die 'failed to create temp file for verbose command output'; fi
+
+# Copy current module cache into temporary directory as basis.
+# Periodically, someone on the project should go through this process without
```

Comment:
There should be a command-line option to skip this initial `rsync` to permit the workflow that this documentation was stating. This comment should live as part of user documentation.

Reply:
Done.

Reply:
It seems to want to remove a lot of things. Maybe we should talk about this.

Reply:
Probably all the modules named "github.com/google/go-cloud/...".

Reply:
Ah, makes sense.